vision_datasets/dataset/vision_as_image_text_dataset.py

```diff
@@ -0,0 +1,61 @@
+import random
+from copy import deepcopy
+
+from ..common import DatasetTypes
+from .base_dataset import BaseDataset
+from ..data_tasks.image_text_matching.manifest import ImageTextMatchingLabelManifest
+
+
+class VisionAsImageTextDataset(BaseDataset):
+    """
+    Consume traditional vision datasets of type
+    [DatasetTypes.IMAGE_CLASSIFICATION_MULTICLASS, DatasetTypes.IMAGE_CLASSIFICATION_MULTILABEL, DatasetTypes.IMAGE_OBJECT_DETECTION], as DatasetTypes.IMAGE_TEXT_MATCHING dataset.
+    For a certain image, negative image-text pairs are generated from the labels that this image does not possess.
+    """
+
+    def __init__(self, dataset: BaseDataset, neg_to_pos_ratio=0, text_aug=None, rnd_seed=0):
+        """
+        Args:
+            dataset: dataset of expected type
+            neg_to_pos_ratio: ratio of negative against positive image text pairs
+            text_aug: a func that augments a string, i.e., a class name, e.g. dog => a photo of dog
+            rnd_seed: random seed for choosing negative class names for negative image text pairs
+        """
+        assert dataset is not None
+        assert dataset.dataset_info.type in [DatasetTypes.IMAGE_CLASSIFICATION_MULTICLASS, DatasetTypes.IMAGE_CLASSIFICATION_MULTILABEL, DatasetTypes.IMAGE_OBJECT_DETECTION]
+        assert neg_to_pos_ratio >= 0
+        dataset_info = deepcopy(dataset.dataset_info)
+        dataset_info.type = DatasetTypes.IMAGE_TEXT_MATCHING
+
+        super().__init__(dataset_info)
+        self._dataset = dataset
+        self._negative_pair_ratio = neg_to_pos_ratio
+        self._text_aug = text_aug or (lambda x: x)
+        self._rand = random.Random(rnd_seed)
```

Comment:
The state of RNG will be changed depending on the order of call of get_item(). To make it deterministic, probably we need to initialize RNG based on image index.